### ISSUE TITLE
Resolving use of both environment.yml and requirements.txt files

### ISSUE THREAD

**Issue by Robinlovelace** [geocompr/py#32]
Currently we have two files defining dependencies: 

https://github.com/geocompr/py/blob/main/data/requirements.txt

and

https://github.com/geocompr/py/blob/main/environment.yml

The fact that we have two versions of the deps can cause issues, e.g. #31.

I think we should standardise and, looking at various help files, am thinking that an environment.yml file that points to the pip deps should be fine. Source: https://stackoverflow.com/a/68164027

```yml
# environment.yml
name: geocompy
dependencies:
  - python>=3.9
  - miniconda3
  - pip
  - pip:
    - -r file:data/requirements.txt
```
However, I note these caveats from https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html :

> Issues may arise when using pip and conda together. When combining conda and pip, it is best to use an isolated conda environment. Only after conda has been used to install as many packages as possible should pip be used to install any remaining software. If modifications are needed to the environment, it is best to create a new environment rather than running conda after pip. When appropriate, conda and pip requirements should be stored in text files.

Thoughts? Another advantage of standardising this could be to automatically update the Docker images when the requirements get updated here, linking to #6

**Comment by Robinlovelace** [comment]
Also the requirements.txt contains way more pkgs and I'm not sure why...

**Comment by jsta** [comment]
Just lurking here, but my recommendation is to drop `requirements.txt` and only use `environment.yml`. IMO `requirements.txt` only makes sense for very simple projects where you don't need environment isolation and can manually configure system deps. Seeing as this project is fairly complex (uses GDAL), best to let conda handle that.

**Comment by Robinlovelace** [comment]
Thanks for the drop-in comment Jem, love how solutions pop-up in open source communities! Heads-up Michael I think the argument has merits but there may be benefits of the requirements.txt approach that I'm not aware of (does not require conda, although as it has GDAL deps there is a need for some system level installation system and conda seems to fit the bill - there's also poetry but that's another story...).

**Comment by michaeldorman** [comment]
Thanks! I'm very not familiar with differences in between `requirements.txt` and `environment.yml`, if the latter is the recommended approach then we should stick with it to keep things simple. In any case it's probably best to have one recommendation of setup per operating system.

**Comment by Robinlovelace** [comment]
Great, let's switch everything to the .yml file. Anything we need to consider before removing requirements.txt?

**Comment by michaeldorman** [comment]
I don't think so...

**Comment by Robinlovelace** [comment]
Great, I'll try fixing it...

**Comment by michaeldorman** [comment]
Thanks Robin

**Comment by Robinlovelace** [comment]
Great, this should close it, please review and merge  if happy @michaeldorman: https://github.com/geocompr/py/pull/34
